eos-enable-zram

```diff
@@ -13,13 +13,79 @@ if [ "$1" -eq 0 ]; then
     exit 0
 fi
 
-ram_size_kb=$(awk '/MemTotal/{print $2}' /proc/meminfo)
+# List and disable any active swap areas
+active_swaps=$(swapon --show=NAME --raw --noheadings)
+echo "Active swap areas before $(basename $0): ${active_swaps}"
+swapoff -a
 
+# Enable zram
+ram_size_kb=$(awk '/MemTotal/{print $2}' /proc/meminfo)
 if [ ! -e /sys/block/zram0 ]; then
     modprobe zram
 fi
 echo $(($ram_size_kb * 2))K > /sys/block/zram0/disksize
-# For now, disable the swap partition if in use
-swapoff -a
 mkswap /dev/zram0
 swapon /dev/zram0
+
+if [ -z "${active_swaps}" ]; then
+    exit 0
+fi
+
+# Find root partition, root partition number, root disk, and table type
+orig_root_part=$(systemctl show -p What sysroot.mount)
+orig_root_part=${orig_root_part#What=}
+root_part=$(readlink -f "${orig_root_part}")
+case ${root_part} in
+    /dev/sd??)
+        root_disk=${root_part%?}
+        ;;
+    /dev/*p[0-9])
+        root_disk=${root_part%p?}
+        ;;
+esac
+
+case ${orig_root_part} in
+    /dev/mapper/endless-image?)
+        root_disk=${orig_root_part%?}
+        ;;
+esac
+
+if [ -z "${root_disk}" ]; then
+    exit 0
+fi
+
+get_last_char() {
+    ret=$1
+    while [ "${#ret}" != "1" ]; do
+            ret=${ret#?}
+    done
+    echo $ret
+}
+partno=$(get_last_char ${root_part})
+pt_label=$(blkid -o value -s PTTYPE $root_disk)
+if [ -z "$pt_label" ]; then
+    echo "blkid -o value -s PTTYPE '$root_disk' failed"
+    exit 0
+fi
+
+# Enable the conditions for endless-repartition.service and
+# eos-firstboot.service so they run on the next boot
+if [ "$pt_label" = "gpt" ]; then
+    attrs=$(sfdisk --part-attrs ${root_disk} ${partno})
+    attrs="GUID:55 ${attrs}"
+    echo "Setting partition attributes \"${attrs}\" on ${root_disk}${partno}"
+    sfdisk --force --part-attrs ${root_disk} ${partno} "${attrs}"
+else
+    # On MBR our marker for expanding the root partition is setting partition
+    # entry 4 with the type 0xdd, which is unused by EOS. Our MBR layouts were
+    # composed of two partitions, id 1 for /boot and id 2 for the root fs, with
```

Comment:
We have MBR layouts today, too, converted from GPT after running `eos-installer` on a BIOS system using `eos-repartition-mbr` (in this repo). These have 1 or 2 non-swap partitions, depending on whether you installed before or after 3.4.0. See https://phabricator.endlessm.com/w/software/x86-64-boot-paths/

I thought we also still use MBR layout on some ARM systems, but I can't find evidence of that in the image builder.

So I think this comment should be updated to make clear that MBR layouts are not just a thing of the past; and we should probably ensure there is not a 4th partition defined in the MBR before we take any of this action.

Reply:
I also don't think there is any scenario where we would have created a 4th partition, but it would be very sad to erase a real partition by writing this marker!